code/_core/obj/structure/interactive/fusion_engine.dm

```diff
@@ -0,0 +1,85 @@
+/*
+So this generator is supposed to be for the station and is a very basic power system.
+As long as it is powered, the station will be powered. The moment it is no longer powered, no power on station.
+No power? no doors, vendors, lights, etc.
+*/
+
+/obj/structure/interactive/fusion_generator
+	name = "fusion generator"
+	desc = "Behold the almighty Atom!"
+	desc_extended = "An engine that fuses atoms together to create energy. Make sure it is always powered!"
+	icon = 'icons/obj/structure/generator.dmi'
+	icon_state = "gen_small_4"
+	density = TRUE
+	anchored = TRUE
+
+	var/stage = 4 //4 to 0, 4 is max power, 0 is no power
+	var/cooldown_time = 0 //the timer basically
+	var/add_cooldown = 300 //how many seconds are checked (5 minutes) 300
+
+/obj/structure/interactive/fusion_generator/Finalize()
+	. = ..()
+	start_thinking(src)
+
+/obj/structure/interactive/fusion_generator/Destroy()
+	stop_thinking(src)
+	. = ..()
+
+/obj/structure/interactive/fusion_generator/think()
+	if(stage <= 0)
+		return TRUE
+	if(world.time < cooldown_time)
+		return TRUE
+	cooldown_time = world.time + SECONDS_TO_DECISECONDS(add_cooldown)
+	if(prob(50)) //50 percent chance to not degrade!
+		return TRUE
+	stage--
+	icon_state = "gen_small_[stage]"
+	update_icon()
+	if(stage <= 0) //if stage is 0
+		SSgenerator_power.station_power = FALSE
+		SSgenerator_power.turn_off_lights()
+	return TRUE
```

Comment:
I think you should incorporate ``return FALSE`` here as this will go into the negatives in terms of stages. Also it's better for performance if it doesn't think.

But I think it would be better if you made a new proc with this loop called /power_think and move the thinking to the new subsystem you made, and make the new subsystem think every 5 minutes or however often you want the system to reset.